pkgs/development/tools/ydiff/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib, pythonPackages, python3Packages, less, patchutils, git
+, subversion, coreutils, which }:
+
+with pythonPackages;
+
+buildPythonApplication rec {
+  pname = "ydiff";
+  version = "1.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0mxcl17sx1d4vaw22ammnnn3y19mm7r6ljbarcjzi519klz26bnf";
+  };
+
+  patchPhase = ''
+    substituteInPlace tests/test_ydiff.py \
+      --replace /bin/rm ${coreutils}/bin/rm \
+      --replace /bin/sh ${stdenv.shell}
+    substituteInPlace Makefile \
+      --replace "pep8 --ignore" "#pep8 --ignore" \
+      --replace "python3 \`which coverage\`" "${python3Packages.coverage}/bin/coverage3" \
+      --replace /bin/sh ${stdenv.shell}
+  '';
+
+  buildInputs = [ docutils pygments ];
+  propagatedBuildInputs = [ less patchutils ];
+  checkInputs = [ coverage coreutils git subversion which ];
+
+  checkPhase = if isPy3k then ''make test3'' else ''make test'';
```

Comment:
Sorry to nitpick, but if it's in a single line, then the `''` quotes aren't needed:
```nix
{
  checkPhase = if isPy3k then "make test3" else "make test";
}
```

:D

Reply:
Indeed, so let's shave off those two extra characters...

Reply:
There is something rewarding with a tool that you can use on the commit itself. 

ydiff -l -s